hw1/02-palindrome.js

```diff
@@ -1,3 +1,40 @@
-const element = document.querySelector("input");
+const integerInputElement = document.getElementById("integer_input");
+const resultElement = document.getElementById("result");
 
-element.addEventListener("input", handleInput);
+integerInputElement.addEventListener("input", handleInput);
+
+function handleInput() {
+  let value = integerInputElement.value;
+  if (value < 0) {
+    resultElement.innerHTML = `<p style="color:red">Enter a positive number</p>`;
+  } else if (value.length === 0) {
+    resultElement.innerHTML = "";
+  } else if (isAlphabetsPresent(value)) {
+    resultElement.innerHTML = `<p style="color:red">Please enter a number.</p>`;
+  } else if (isPalindrome(value)) {
+    resultElement.innerHTML = `<p style="color:green">Yes. This is a palindrome!</p>`;
+  } else {
+    resultElement.innerHTML = `<p style="color:red">No. Try again.</p>`;
+  }
+}
+
+function isAlphabetsPresent(value) {
+  var regExp = /[a-zA-Z]/g;
```

Comment:
Cool work with writing the regex but I am not able to give alphabet string to the input field, maybe this is from my end if it's working fine for others!

Reply:
This is because the input type is number so we cant enter an alphabet. I still went ahead and added a check to see if we encounter any alphabets in the input field.